linkis-computation-governance/linkis-entrance/src/main/java/org/apache/linkis/entrance/restful/EntranceMetricRestfulApi.java

```diff
@@ -0,0 +1,104 @@
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.linkis.entrance.restful;
+
+import org.apache.linkis.common.conf.Configuration;
+import org.apache.linkis.entrance.EntranceServer;
+import org.apache.linkis.entrance.annotation.EntranceServerBeanAnnotation;
+import org.apache.linkis.entrance.execute.EntranceJob;
+import org.apache.linkis.manager.label.entity.engine.EngineTypeLabel;
+import org.apache.linkis.manager.label.utils.LabelUtil;
+import org.apache.linkis.server.Message;
+import org.apache.linkis.server.utils.ModuleUserUtils;
+
+import org.apache.commons.lang3.StringUtils;
+
+import org.springframework.web.bind.annotation.*;
+
+import javax.servlet.http.HttpServletRequest;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@RestController
+@RequestMapping(path = "/entrance/api/metrics")
+public class EntranceMetricRestfulApi {
+
+    private EntranceServer entranceServer;
+
+    private static final Logger logger = LoggerFactory.getLogger(EntranceMetricRestfulApi.class);
+
+    @EntranceServerBeanAnnotation.EntranceServerAutowiredAnnotation
+    public void setEntranceServer(EntranceServer entranceServer) {
+        this.entranceServer = entranceServer;
+    }
+
+    @RequestMapping(path = "/taskinfo", method = RequestMethod.GET)
+    public Message status(
+            HttpServletRequest req,
+            @RequestParam(value = "user", required = false) String user,
+            @RequestParam(value = "creator", required = false) String creator,
+            @RequestParam(value = "engineTypeLabel", required = false)
+                    String engineTypeLabelValue) {
+        String userName = ModuleUserUtils.getOperationUser(req, "taskcount");
+        String queryUser = user;
+        if (!Configuration.isAdmin(userName)) {
+            if (StringUtils.isBlank(queryUser)) {
+                queryUser = userName;
+            } else if (!userName.equalsIgnoreCase(queryUser)) {
+                return Message.error(
+                        "Non-administrators cannot view other users' task information");
+            }
+        }
+        String filterWords = creator;
+        if (StringUtils.isNoneBlank(filterWords) && StringUtils.isNoneBlank(queryUser)) {
```

Comment:
nit: for single parameter, I think `StringUtils.isNotBlank` is better.

Reply:
> nit: for single parameter, I think `StringUtils.isNotBlank` is better.

yes